src/storage/src/hummock/compactor/shared_buffer_compact.rs

```diff
@@ -397,57 +369,9 @@ pub async fn merge_imms_in_memory(
 
             // Reset state before moving onto the new table key
             table_key_versions = vec![];
-            table_key_last_delete_epoch = HummockEpoch::MAX;
-
-            // Record range tombstones if any
-            let target_extended_user_key = PointRange::from_user_key(
-                full_key_tracker.latest_full_key.user_key.as_ref(),
-                false,
-            );
-            while del_iter.is_valid() && del_iter.key().le(&target_extended_user_key) {
-                let event_key = del_iter.key().to_vec();
-                del_iter.next().await?;
-                monotonic_tombstone_events.push(MonotonicDeleteEvent {
-                    event_key,
-                    new_epoch: del_iter.earliest_epoch(),
-                });
-            }
-        }
-        let earliest_range_delete_which_can_see_key =
-            del_iter.earliest_delete_since(epoch_with_gap.pure_epoch());
-        if value.is_delete() {
-            table_key_last_delete_epoch = epoch_with_gap.pure_epoch();
-        } else if earliest_range_delete_which_can_see_key < table_key_last_delete_epoch {
```

Comment:
When there is no range delete in del_iter, the earliest_delete_since will return u64::MAX, and then we will never reach this branch.